dagfactory/listeners/runtime_event.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import annotations
+
+from airflow.listeners import hookimpl
+from airflow.models.dag import DAG
+from airflow.models.dagrun import DagRun
+from airflow.models.taskinstance import TaskInstance
+from airflow.utils.log.logging_mixin import LoggingMixin
+from airflow.utils.state import TaskInstanceState
+from packaging import version
+
+from dagfactory import telemetry
+
+try:
+    from airflow.version import version as AIRFLOW_VERSION
+except ImportError:
+    from airflow import __version__ as AIRFLOW_VERSION
+
+log = LoggingMixin().log
+
+
+class EventStatus:
+    SUCCESS = "success"
+    FAILED = "failed"
+
+
+class EventType:
+    DAG_RUN = "dag_run"
+    TASK_INSTANCE = "task_instance"
+
+
+def is_dagfactory_dag(dag: DAG | None = None):
+    if "dagfactory" in dag.tags:
+        return True
+    return False
+
+
+@hookimpl
+def on_dag_run_success(dag_run: DagRun, msg: str):
+    if not is_dagfactory_dag(dag_run.get_dag()):
+        return
+    additional_telemetry_metrics = {
+        "dag_hash": dag_run.dag_hash,
```

Comment:
What does the `dag_hash` contain? May it contain any personal/company-identifiable data?

Reply:
This is an md5 hash, so this will just be a 32bit char and not any data example hash: d151d1fa2f03270ea116cc7494f2c591

Reply:
Regarding the reversibility of MD5:
https://md5.gromweb.com/

I don't think that would be an issue. Still, we could consider r using another way of generating hashes, like using more robust algorithms ([SHA-2](https://en.wikipedia.org/wiki/SHA-2), [Whirlpool](https://en.wikipedia.org/wiki/Whirlpool_%28cryptography%29), etc), combining algorithms, and using a "[salt](https://en.wikipedia.org/wiki/Salt_%28cryptography%29)".

Reply:
Yes, the salt-based hash is more secure. But since Airflow calculates this hash, I think we can keep it for now that way we do not have to maintain it. but please let me know if you want me to change to some other hashing algorithm